network/p2p/cache/node_blocklist_distributor.go

```diff
@@ -0,0 +1,38 @@
+package cache
+
+import (
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/network/p2p"
+)
+
+type OnNodeBlockListConsumer = func(blockList flow.IdentifierList)
```

Comment:
please add a concurrency safe & non-blocking notice here as well

Reply:
I'd prefer to avoid using abstract functions to specify consumers. See [my comment below](https://github.com/onflow/flow-go/pull/3762#issuecomment-1382490758) for additional details.